File: Frontend/src/types/rental.ts
export interface CreateRentalRequest {
  customerId: string;
  carId: string;
  startDate: string;
  endDate: string;
}

export interface RentalResponse {
  id: string;
  customerId: string;
  carId: string;
  startDate: string;
  endDate: string;
  returnedAt: string | null;
  status: string;
  createdAt: string;
}

File: Api/Models/Rental.cs
namespace Api.Models;

public class Rental
{
    public Guid Id { get; set; }
    public Guid CustomerId { get; set; }
    public Guid CarId { get; set; }
    public DateTimeOffset StartDate { get; set; }
    public DateTimeOffset EndDate { get; set; }
    public DateTimeOffset? ReturnedAt { get; set; }
    public RentalStatus Status { get; set; }
    public int? KilometersDriven { get; set; }
    public DateTimeOffset CreatedAt { get; set; }

    public Customer Customer { get; set; } = null!;
    public Car Car { get; set; } = null!;
}
